pingcap/tidb
# server: remove error stack in log file when parsing failed

Remove useless error stacks in logs for parseing errors. Solves issue #15017.



### What problem does this PR solve? 
Adds a function that will determine whether or not to skip logging an error's stack trace after
a dispatched command fails. For now, it only will skip if the error is a parse error, but 
more errors can be added to skip as needed.

### What is changed and how it works?
Add check before logging a failed command's full stack trace:
```

doLogStackTrace := !shouldSkipStackTrace(
if doLogStackTrace {
...
```
```
func shouldSkipStackTrace(err error) bool {
	return parser.ErrParse.Equal(err)
}
```

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
Before: 
```
[2020/02/29 11:39:03.030 -07:00] [WARN] [conn.go:698] ["command dispatched failed"] [conn=1] [connInfo="id:1, addr:127.0.0.1:47146 status:10, collation:utf8_general_ci, user:root"] [command=Query] [status="inTxn:0, autocommit:1"] [sql="show tablsaa"] [txn_mode=OPTIMISTIC] [err="[parser:1064]You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 12 near \"tablsaa\" \ngithub.com/pingcap/errors.AddStack\n\t/home/peter/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20190809092503-95897b64e011/errors.go:174\ngithub.com/pingcap/parser
...
```
After: 
```
[2020/02/29 12:54:31.122 -07:00] [WARN] [session.go:1095] ["parse SQL failed"] [conn=1] [error="line 1 column 12 near \"tablsaa\" "] [SQL="show tablsaa"]
[2020/02/29 12:54:31.122 -07:00] [ERROR] [misc.go:121] ["syntax error"] [error="line 1 column 12 near \"tablsaa\" "]
```

Code changes

Side effects


Related changes


Release note

 - Don't log error stack when parsing failed

PS: First time contributing to this project so let me know if I am missing anything. Thanks!
